src/components/shared/thread/ActivityGraphCanvas.js

```diff
@@ -196,10 +221,12 @@ export class ActivityGraphCanvas extends React.PureComponent<CanvasProps> {
   render() {
     const { className, trackName, onClick } = this.props;
     return (
-      <canvas className={className} ref={this._canvas} onClick={onClick}>
-        <h2>Activity Graph for {trackName}</h2>
-        <p>This graph shows a visual chart of thread activity.</p>
-      </canvas>
+      <InView onChange={this._observerCallback}>
```

Comment:
We may want to specify the `root` parameter to the ref of the container. But this looks complicated to get in the first place, and maybe not worth it. This is for a follow-up if we want this, not for this patch.

Reply:
Yeah, I didn't want to do it right now since this works fine too. It could be a follow-up.